src/FluentMigrator.Runner/MigrationRunner.cs

```diff
@@ -280,7 +259,6 @@ public virtual void ApplyMigrationUp(IMigrationInfo migrationInfo, bool useTrans
                 _alreadyOutputPreviewOnlyModeWarning = true;
             }
 
-            if (!migrationInfo.IsAttributed() || !VersionLoader.VersionInfo.HasAppliedMigration(migrationInfo.Version))
```

Comment:
BTW: You may create a private function that takes an additional boolean that might skip the validation like e.g.:

```
 if (!skipAttributedCheck && (!migrationInfo.IsAttributed() || !VersionLoader.VersionInfo.HasAppliedMigration(migrationInfo.Version)))
```

You'd then be able to call this private function from all other code paths, but with `skipAttributedCheck` set to `true`.